packages/wallet/backend/src/rapyd/rapyd-client.ts

```diff
@@ -328,19 +391,30 @@ export class RapydClient implements IRapydClient {
     }
   }
 
-  private async request<T = unknown>(
+  private async request<
+    T extends AnyZodObject | ZodEffects<AnyZodObject> | z.ZodTypeAny
+  >(
     method: 'get' | 'post' | 'put',
     url: string,
-    body?: string
-  ) {
+    body?: string,
+    zodResponseSchema?: T
+  ): Promise<RapydResponse<z.infer<T>>> {
     const headers = this.getRapydRequestHeader(method, url, body ?? '')
     try {
-      const res = await axios<T>({
+      const res = await axios<z.infer<T>>({
         method,
         url: `${this.deps.env.RAPYD_API}/${url}`,
         ...(body && { data: body }),
         headers
       })
+
+      if (zodResponseSchema) {
+        await validateRapydResponse(
+          zodResponseSchema as AnyZodObject | ZodEffects<AnyZodObject>,
+          res.data
```

Comment:
I think here should be `res.data.data` as one `data` is from axios and another one is rapyd response